### PR TITLE
added support for Kimax U-25AWF

### PR DESCRIPTION
**Pull request by danielkucera**
rewrite from https://forum.openwrt.org/viewtopic.php?pid=305643

### CODE REVIEW

**Comment by pepe2k** [comment]
Please, fix your commit based on the information about submitting patches: [here](https://lede-project.org/docs/guide-developer/the-source-code#submitting_patches) and [here](https://github.com/lede-project/source/pull/592). In current shape it cannot be accepted.

File: target/linux/ramips/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -293,6 +293,10 @@ re6500)
 	ucidef_set_led_default "power" "power" "$board:white:power" "1"
 	ucidef_set_led_default "wifi" "wifi" "$board:orange:wifi" "1"
 	;;
+u25awf)
+        set_wifi_led "u25awf:red:wifi"
+        ucidef_set_led_netdev "eth" "eth" "u25awf:green:lan" "eth0"
+        ;;
 rp-n53)
```
Comment (blogic) [inline target/linux/ramips/base-files/etc/board.d/01_leds]: wrong alphabetical ordering

File: target/linux/ramips/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -134,6 +134,7 @@ ramips_setup_interfaces()
 	atp-52b|\
 	awm002-evb|\
 	awm003-evb|\
+        u25awf|\
 	c20i|\
```
Comment (blogic) [inline target/linux/ramips/base-files/etc/board.d/02_network]: wrong ordering and bad indenting

File: target/linux/ramips/base-files/etc/diag.sh [hunk 3]
```diff
@@ -171,6 +171,9 @@ get_status_led() {
 	nw718)
 		status_led="$board:amber:cpu"
 		;;
+        u25awf)
+                status_led="u25awf:red:wifi"
+                ;;
 	newifi-d1)
```
Comment (blogic) [inline target/linux/ramips/base-files/etc/diag.sh]: wrong ordering and bad indenting

File: target/linux/ramips/base-files/lib/ramips.sh [hunk 4]
```diff
@@ -511,6 +511,9 @@ ramips_board_detect() {
 	*"WF-2881")
 		name="wf-2881"
 		;;
+        *"U-25AWF")
+                name="u25awf"
+                ;;
 	*"WHR-1166D")
```
Comment (blogic) [inline target/linux/ramips/base-files/lib/ramips.sh]: wrong ordering and bad indenting

File: target/linux/ramips/base-files/lib/upgrade/platform.sh [hunk 5]
```diff
@@ -151,6 +151,7 @@ platform_check_image() {
 	witi|\
 	wizfi630a|\
 	wl-330n|\
+        u25awf |\
 	wl-330n3g|\
```
Comment (blogic) [inline target/linux/ramips/base-files/lib/upgrade/platform.sh]: wrong ordering and bad indenting

File: target/linux/ramips/dts/U25AWF.dts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+/include/ "mt7620n.dtsi"
+
+/ {
+       compatible = "u25awf", "ralink,mt7620n-soc";
+       model = "Kimax U-25AWF";
+
+       palmbus@10000000 {
+                               gpio0: gpio@600 {
+                        status = "okay";
```
Comment (blogic) [inline target/linux/ramips/dts/U25AWF.dts]: bad indenting

File: target/linux/ramips/dts/U25AWF.dts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+                };
+
+                               gpio1: gpio@638 {
+                        status = "okay";
```
Comment (blogic) [inline target/linux/ramips/dts/U25AWF.dts]: bad indenting

File: target/linux/ramips/dts/U25AWF.dts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+                };
+
+                gpio2: gpio@660 {
+                        status = "okay";
+                };
+
+                gpio3: gpio@688 {
+                        status = "okay";
+                };
+
```
Comment (blogic) [inline target/linux/ramips/dts/U25AWF.dts]: the whole file is badly indented. please use tabs and not spaces

**Comment by danielkucera** [comment]
Ok, I'll submit a new one.

**Comment by danielkucera** [comment]
I just commit-ed a new version.

**Comment by bobafetthotmail** [comment]
Looks good, but it lacks 

Signed-off-by: name surname <email@email.com>

Please add it.

**Comment by danielkucera** [comment]
Oh, I forgot... What about now?

File: target/linux/ramips/dts/U25AWF.dts [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+			status="okay";
+		};
+
+		spi@b00{
+			status="okay";
+
+			m25p80@0{
+				#address-cells=<1>;
+				#size-cells=<1>;
+				compatible="mx25l12805d";
```
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: compatible = "jedec,spi-nor";

File: target/linux/ramips/dts/U25AWF.dts [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+				reg=<00>;
```
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: reg=<0>;

File: target/linux/ramips/dts/U25AWF.dts [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+				linux,modalias="m25p80","m25p128";
```
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: Drop this line. It was only required for kernels < 4.4

File: target/linux/ramips/dts/U25AWF.dts [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+/include/"mt7620n.dtsi"
+
+/{
+	compatible="u25awf","ralink,mt7620n-soc";
+	model="KimaxU-25AWF";
+
+	palmbus@10000000{
+		gpio0:gpio@600{
+			status="okay";
```
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: please use whitespaces around the equal sign. Without the equal signs it is kinda hard to read.

The same applies to all ```{```. They should have a leading whitespace.

The same applies to all ```:```. They should have a trainling whitespace.

Have a look how it is done in the other dts files.

File: target/linux/ramips/dts/U25AWF.dts [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+		};
+
+		gpio1:gpio@638{
```
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: drop this node. as far as i can see, gpio1 isn't used in the dts.

File: target/linux/ramips/dts/U25AWF.dts (hunk 12 above)
Comment (mkresin) [inline target/linux/ramips/dts/U25AWF.dts]: You can drop the whole gpio0 node. it is already enabled in mt7620n.dtsi.

**Comment by mkresin** [comment]
Please provide a bootlog after you done all requested changes. Some of the spotted mistakes make it quite obvious that it isn't runtime tested and maybe even not compile tested.

**Comment by danielkucera** [comment]
guys, I don't have serial access to my device (it's kinda hard to solder) so I'm not going to experiment with these changes.

**Comment by Mynzer** [comment]
Hi, good day.

I'm interested in that this device gets full lede support in the future, currently my device is running a compiled lede 17.01.2 version with @danielkucera patch for openwrt

https://github.com/openwrt/openwrt/pull/380/files

I'm agree with @danielkucera that is very hard to solder serial access to this device, so in order to avoid brick my device, I've compiled a lede version with unlocked u-boot partition and I've change stock u-boot to breed bootloader for easy recovery this device.

in case that @danielkucera is interested, y could test any firmware that he compile and provide boot logs, or i could provide the steps and files that i use to change the stock u-boot so that @danielkucera could make any test in his device.

Greetings and thanks for all the hard work

**Comment by danielkucera** [comment]
Hi Mynzer,

I already have my device in "production" so I'm not able to develop on in. But I can try to apply requested changes if you are willing to test them.

**Comment by danielkucera** [comment]
Rebased against current master, new PR: #1402

**Comment by danielkucera** [comment]
@Mynzer can you please test branch u25awf https://github.com/danielkucera/source/tree/u25awf and provide bootlog?

**Comment by Mynzer** [comment]
Ok, when i try to compile from https://github.com/danielkucera/source/tree/u25awf branch, i've got the following error
[CompileError.txt](https://github.com/lede-project/source/files/1363584/CompileError.txt)

**Comment by danielkucera** [comment]
try to pull it now

**Comment by danielkucera** [comment]
@Mynzer there were more errors, please try current version and respond to issue #1402  from now on.

**Comment by Mynzer** [comment]
Ok, I'm in the middle of a previous compilation, I'm going to cancel all, pull again from your branch and report on #1402

**Comment by danielkucera** [comment]
you don't need to make clean when testing updated version. just `git pull` and `make`

**Comment by Mynzer** [comment]
Ok, I see this message to late, next updates will use `git pull`, by the way, I'm ready to provide bootlog when you apply the new change request

**Comment by danielkucera** [comment]
So what's your status now?

Dňa 7. 10. 2017 12:10 dopoludnia používateľ "Mynzer" <
notifications@github.com> napísal:

**Pull request closed without merging**